importlib_resources/tests/test_reader.py

```diff
@@ -67,7 +69,11 @@ def test_join_path(self):
             os.path.join('namespacedata01', 'binary.file'),
         )
         self.assertEqual(
-            str(path.joinpath('subdirectory'))[len(prefix) + 1 :],
+            str(path.joinpath('subdirectory')._paths[0])[len(prefix) + 1 :],
+            os.path.join('namespacedata01', 'subdirectory'),
+        )
+        self.assertEqual(
+            str(path.joinpath('subdirectory')._paths[1])[len(prefix) + 1 :],
```

Comment:
If I'm reading this correctly, the meaning of this test has changed because previously `namespacedata01` did not contain a subdirectory and now it does, along with `data01`, which also has that subdirectory. Because of this complication, the test now is a lot more complicated, reaching into private members of the MultiplexedPath and extracting values by index (0, 1). I'd like to avoid adding that complication to the test. 

I can think of a few options here:

- rename `subdirectory` to something else so it doesn't overlap with data01.
- use another assertion here about what's to be found in the sub directory.
- make separate assertions about the inner paths without using private members.

Reply:
Addressed in 6c9b19006cbedd5237ee37bc1cdfd41d85ef0d30.